intake_esm/cmip.py

```diff
@@ -132,39 +129,58 @@ def _get_file_attrs(self, filepath):
-        )
-        fileparts.update(f)
-
-        parent = os.path.dirname(filepath).strip('/')
-        parent_split = parent.split(f"/{fileparts['source_id']}/")
-        part_1 = parent_split[0].strip('/').split('/')
+        For Zarr-ified CMIP6, intake-esm assumes a very strict format
 
-        grid_label = parent.split(f"/{fileparts['variable_id']}/")[1].strip('/').split('/')[0]
+        <activity_id>/
+            <institution_id>/
+                <source_id>/
+                    <experiment_id>/
+                        <member_id>/
+                            <table_id>/
+                                <variable_id>/
+                                    <grid_label>
 
-        fileparts['file_dirname'] = os.path.dirname(filepath) + '/'
-        fileparts['grid_label'] = grid_label
-        fileparts['activity_id'] = part_1[-2]
-        fileparts['institution_id'] = part_1[-1]
-
-        version_regex = r'v\d{4}\d{2}\d{2}|v\d{1}'
-        version = _extract_attr_with_regex(parent, regex=version_regex) or 'v0'
-        fileparts['version'] = version
-        fileparts['mip_era'] = config.get('collections.cmip6.mip_era')
+        """
+        keys = list(set(self.columns))
+        fileparts = {key: None for key in keys}
+        fileparts['path'] = storepath
+
+        if storepath.endswith('.nc'):
+            file_basename = os.path.basename(storepath)
+            filename_template = '{variable_id}_{table_id}_{source_id}_{experiment_id}_{member_id}_{grid_label}_{time_range}.nc'
+            gridspec_template = (
+                '{variable_id}_{table_id}_{source_id}_{experiment_id}_{member_id}_{grid_label}.nc'
+            )
+
+            f = _reverse_filename_format(
+                file_basename,
+                filename_template=filename_template,
+                gridspec_template=gridspec_template,
+            )
+
+            fileparts.update(f)
+
+            parent = os.path.dirname(storepath).strip('/')
+            parent_split = parent.split(f"/{fileparts['source_id']}/")
+            part_1 = parent_split[0].strip('/').split('/')
+
+            grid_label = parent.split(f"/{fileparts['variable_id']}/")[1].strip('/').split('/')[0]
+            fileparts['grid_label'] = grid_label
+            fileparts['activity_id'] = part_1[-2]
+            fileparts['institution_id'] = part_1[-1]
+
+        else:
+            # For Zarr-ified CMIP6 data, we assume a very strict format:
+            # See docstring notes
+            parts = storepath.strip('/').split('/')[-8:]
+            fileparts['activity_id'] = parts[-8]
+            fileparts['institution_id'] = parts[-7]
+            fileparts['source_id'] = parts[-6]
+            fileparts['experiment_id'] = parts[-5]
+            fileparts['member_id'] = parts[-4]
+            fileparts['table_id'] = parts[-3]
+            fileparts['variable_id'] = parts[-2]
+            fileparts['grid_label'] = parts[-1]
```

Comment:
Hard coding of these relationships makes me worried. That means any change to the catalog file format requires a change to intake-esm code.

Would be better to factor this out into a configuration file.

Reply:
> Would be better to factor this out into a configuration file.

I concur! This was a temporary hack. I am in favor of your suggestion of allowing the user to specify which template to use and then rendering the template.

Reply:
@rabernat, @matt-long 

How about adding a template key in the input yaml file?

```yaml
  name: GLADE-CMIP6
  collection_type: cmip6
  data_sources:
   GLADE-DATA:
    template: <variable_id>_<table_id>_<source_id>_<experiment_id >_<member_id>_<grid_label>[_<time_range>].nc
    locations:
      - name: CMIP6
        loc_type: posix
        direct_access: True 
        urlpath: /glade/collections/cmip/CMIP6
        exclude_dirs: ['*/files/*', '*/latest/*']
        data_format: netcdf
```

```yaml
 name: PANGEO-CMIP6
  collection_type: cmip6
  data_sources:
    CMIP6:
      temperate: <activity_id>/<institution_id>/<experiment_id>/<member_id >/<table_id>/<variable_id>/<grid_label>
      locations:
        - name: Google Storage
          loc_type: input-file
          direct_access: True
          urlpath: ./storelist.txt
          data_format: zarr
```